### PR TITLE
[types] Implement views for various system events

### PR DESCRIPTION
**Pull request by runtian-zhou**
## Motivation

Implement code for parsing various system events.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

(Share your test plan here. If you changed code, please provide us with clear instructions for verifying that your changes work.)

### CODE REVIEW

**Comment by runtian-zhou** [comment]
Question for @christinacelee how can I test the json-rpc impl over there?

**Comment by christinacelee** [comment]
overall lgtm, but can you locally test those endpoints work? either json rpc unit tests, or with local curl requests

File: types/src/account_config/events/new_block.rs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    pub fn proposed_time(&self) -> u64 {
+        self.time_micro_seconds
+    }
+
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self> {
+        lcs::from_bytes(bytes).map_err(Into::into)
+    }
+
+    pub fn event_key() -> EventKey {
+        EventKey::new_from_address(&CORE_CODE_ADDRESS, 14)
```
Comment (christinacelee) [inline types/src/account_config/events/new_block.rs:40]: I'm seeing duplicate declarations of this: https://github.com/libra/libra/blob/master/types/src/block_metadata.rs#L75-L77 can we try to clean this up, and maybe other event keys? maybe we need a better way to organize events and their keys too, long term

**Comment by runtian-zhou** [comment]
Just did a local test myself:
1. run `cargo run -p libra-swarm -- -s` to start a local json-rpc server.
2. Execute `curl http://localhost:59826   -H 'Content-Type: application/json' -X POST  --data '{"jsonrpc":"2.0","method":"get_events","params": ["0e000000000000000000000000000000000000000a550c18", 0, 10], "id":1}'
` to fetch the NewBlock event. Results are:
```
{
  "id": 1,
  "jsonrpc": "2.0",
  "result": [
    {
      "data": {
        "proposed_time": 1590087156731372,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 1,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 0,
      "transaction_version": 1
    },
    {
      "data": {
        "proposed_time": 1590087157039105,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 2,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 1,
      "transaction_version": 2
    },
    {
      "data": {
        "proposed_time": 1590087157340510,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 3,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 2,
      "transaction_version": 3
    },
    {
      "data": {
        "proposed_time": 1590087157708854,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 4,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 3,
      "transaction_version": 4
    },
    {
      "data": {
        "proposed_time": 1590087158109303,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 5,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 4,
      "transaction_version": 5
    },
    {
      "data": {
        "proposed_time": 1590087158488740,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 6,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 5,
      "transaction_version": 6
    },
    {
      "data": {
        "proposed_time": 1590087158804079,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 7,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 6,
      "transaction_version": 7
    },
    {
      "data": {
        "proposed_time": 1590087159106493,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 8,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 7,
      "transaction_version": 8
    },
    {
      "data": {
        "proposed_time": 1590087159409198,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 9,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 8,
      "transaction_version": 9
    },
    {
      "data": {
        "proposed_time": 1590087159714410,
        "proposer": "5942d356f114089d4a46f2f3b0b15b52",
        "round": 10,
        "type": "newblock"
      },
      "key": "0e000000000000000000000000000000000000000a550c18",
      "sequence_number": 9,
      "transaction_version": 10
    }
  ]
}
```

**Pull request closed without merging**